app/routes/problem_routes.py

```diff
@@ -27,3 +28,33 @@ def get_problem_by_id(problem_id):
     
     return jsonify(problem), 200
 
+# Used by Leetcode bot to set a problem as solved
+@bp.route('/<int:problem_id>/solved', methods=['PUT'])
+def mark_problem_solved(problem_id):
+    if not problem_id:
+        return jsonify({'error': 'Problem ID not provided'}), 400
+
+    success = firestore_wrapper.update_last_asked_timestamp(problem_id, datetime.now())
+
+    if not success:
+        return jsonify({'error': f'Cannot mark {problem_id} as solved'}), 400
+    
+    return jsonify({'message': f'Marked {problem_id} as solved'}), 200
+
+
+# Used by Leetcode bot to get a random problem as per user request
+@bp.route('/<difficulty>/request', methods=['GET'])
+def get_problem_user_request(difficulty):
+    if not difficulty:
+        return jsonify({'error': 'Difficulty not provided'}), 400
+    
+    if difficulty not in ['easy', 'medium', 'hard']:
+        return jsonify({'error': 'Invalid difficulty'}), 400
+
+    # select_problem_at_random() = no cache
+    problem = problem_manager.select_problem_at_random(difficulty)
+    if not problem:
+        return jsonify({'error': 'No problem found'}), 404
+    
+    return jsonify(problem), 200
+
```

Comment:
Similar problem as before, this needs to have include the user's discord id. We may not use it here yet though